include/utils/error/error_filter.hpp

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#ifndef UTILS_ERROR_FILTER_HPP
+#define UTILS_ERROR_FILTER_HPP
+
+#include <utils/error/error.hpp>
+#include <utils/types.hpp>
+
+#include <variant>
+
+namespace Everest {
+namespace error {
+
+///
+/// \brief This filter is used to filter errors by their state.
+///
+using StateFilter = State;
+std::string state_filter_to_string(const StateFilter& s);
+StateFilter string_to_state_filter(const std::string& s);
+
+///
+/// \brief This filter is used to filter errors by their origin.
+///
+using OriginFilter = ImplementationIdentifier;
+
+///
+/// \brief This filter is used to filter errors by their type.
+///
+using TypeFilter = ErrorType;
+
+///
+/// \brief This filter is used to filter errors by their severity.
+///
+enum class SeverityFilter {
+    LOW_GE,    // greater or equal LOW
```

Comment:
You should probably use a [doxygen member documentation comment here ](https://www.doxygen.nl/manual/docblocks.html#memberdoc) like:
```suggestion
    LOW_GE,    ///< greater or equal LOW
```